src/schema/schema_stat.c

```diff
@@ -53,6 +53,60 @@ err:	__wt_scr_free(session, &buf);
 }
 
 /*
+ * __curstat_size_only --
+ *	 For very simple tables we can avoid getting table handles if
+ *	 configured to only retrieve the size. It's worthwhile because
+ *	 workloads that create and drop a lot of tables can put a lot of
+ *	 pressure on the table list lock.
+ */
+static int
+__curstat_size_only(WT_SESSION_IMPL *session,
+    const char *uri, bool *was_fast,WT_CURSOR_STAT *cst)
+{
+	WT_CONFIG cparser;
+	WT_CONFIG_ITEM ckey, colconf, cval;
+	WT_DECL_RET;
+	WT_ITEM namebuf;
+	wt_off_t filesize;
+	char *tableconf;
+
+	WT_CLEAR(namebuf);
+	*was_fast = false;
+
+	if (!F_ISSET(cst, WT_CONN_STAT_SIZE))
+		return (0);
```

Comment:
I understand that this test will probably be inlined, but I think it would be clearer to check in the calling code.